Assets/Scripts/Models/Scheduler/ScheduledEvent.cs

```diff
@@ -260,6 +260,21 @@ public void Stop()
             RepeatsForever = false;
         }
 
+        public void SetCooldown(float newCooldown)
+        {
+            if (newCooldown < TimeToWait)
+            {
+                // Decrement TimeToWait by the amount of time that has passed since the last time this event fired
+                TimeToWait -= Cooldown - TimeToWait;
```

Comment:
Why not `TimeToWait = newCooldown - (Cooldown - TimetoWait)`?? I'm getting confused by this.
Also, please write unit tests speccing out the desired behaviour for this function. :grinning: